ts/index.ts

```diff
@@ -35,15 +35,17 @@ class ElementaryRunner implements CompileOK {
     opts: CompilerOpts) {
 
     this.codeMap = {};
-    const whitelistCode = opts.whitelistCode;
-    if (whitelistCode) {
-      Object.keys(whitelistCode).forEach((moduleName) => {
-        this.codeMap[moduleName] = eval(`(${whitelistCode[moduleName]})`);
-      });
+    const config: string = `{
+      getRunner: runtime.getRunner,
+      stopifyArray: runtime.stopifyArray,
+      stopifyObjectArrayRecur: runtime.stopifyObjectArrayRecur
+    }`;
+    for (const moduleName in opts.whitelistCode) {
```

Comment:
There is general consensus that `for .. in` is a bad idea. The Mozilla article describes it mildly:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

This code assumes that no non-enumerable properties exist on `Object.prototype`, which happens to be true, but is not a very safe assumption.

Besides, in this code base (and in our research group), loops are considered bad design. Use a better-abstraction, such as higher-order functions.

Reply:
I'm aware of the concerns, as I made a comment in the PR notes.
I will change to use HOF.